infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -131,9 +241,98 @@ private boolean registryTableHasNewEntries() {
+            registrationResults.forEach(entry -> {
+                String clientName = entry.getKey().getClientName();
+                ReplicationModel model = entry.getPayload().getModel();
+                ReplicationSubscriber subscriber = ReplicationSubscriber.newBuilder()
+                        .setClientName(clientName).setModel(model).build();
+                registeredSubscribers.add(subscriber);
+            });
+
+            List<CorfuStoreEntry<ClientDestinationInfoKey, DestinationInfoVal, Message>> groupConfigResults =
+                    txn.executeQuery(TableRegistry.getFullyQualifiedTableName(
+                            CORFU_SYSTEM_NAMESPACE, LR_MODEL_METADATA_TABLE), record -> true);
+            groupConfigResults.forEach(entry -> {
+                String clientName = entry.getKey().getClientName();
+                ReplicationModel model = entry.getKey().getModel();
+                ReplicationSubscriber subscriber = ReplicationSubscriber.newBuilder()
+                        .setClientName(clientName).setModel(model).build();
+                String groupName = entry.getKey().getGroupName();
+                if (registeredSubscribers.contains(subscriber)) {
+                    groupDestinationsMap.put(groupName, new HashSet<>(entry.getPayload().getDestinationIdsList()));
+                } else {
+                    log.warn("Subscriber {} not registered, but found its group config in client table: {}",
+                            subscriber, groupName);
+                }
+            });
+
+            logTail = txn.commit();
+        }
+        return logTail;
+    }
+
+    /**
+     * Callback method which is invoked when the client config listener receives update for new client registration.
+     */
+    public void onNewClientRegister(ReplicationSubscriber subscriber) {
+        if (registeredSubscribers.contains(subscriber)) {
+            log.warn("Client {} with model {} already registered!", subscriber.getClientName(), subscriber.getModel());
+            return;
+        }
+        registeredSubscribers.add(subscriber);
+    }
+
+    /**
+     * Callback method which is invoked when the client config listener receives update for adding/removing
+     * Sinks from a logical group.
+     */
+    public void onGroupDestinationsChange(ReplicationSubscriber subscriber, String logicalGroup,
+                                          List<String> destinations, Set<LogReplicationSession> impactedSessions) {
+        if (!registeredSubscribers.contains(subscriber)) {
+            log.error("Client {} has not registered with replication model {}. Please register first.",
+                    subscriber.getClientName(), subscriber.getModel());
+            return;
+        }
+
+        Set<String> updatedDestinations = new HashSet<>(destinations);
+        Set<String> impactedDestinations = groupDestinationsMap.get(logicalGroup);
```

Comment:
In the case we are adding to a group that is not initialized yet this would yield null, maybe we can use getOrDefault here.